src/states/common/base.rs

```diff
@@ -258,33 +294,88 @@ pub trait Base: Display {
     }
 
     fn send_direct_message(&mut self, dst_id: &PublicId, content: DirectMessage) {
+        let next_id = self.network_service_mut().next_msg_id();
+        self.send_direct_message_with_id(dst_id, content, next_id);
+    }
+
+    fn send_direct_message_with_id(
+        &mut self,
+        dst_id: &PublicId,
+        content: DirectMessage,
+        msg_id: u64,
+    ) {
         let message = if let Ok(message) = self.to_signed_direct_message(content) {
             message
         } else {
             return;
         };
 
-        self.send_message(dst_id, message)
+        self.send_message_with_id(dst_id, message, msg_id)
     }
 
     fn send_message(&mut self, dst_id: &PublicId, message: Message) {
-        let conn_info = match self.peer_map().get_connection_info(dst_id) {
-            Some(conn_info) => conn_info.clone(),
-            None => {
-                error!(
-                    "{} - Failed to send message to {:?} - not connected",
-                    self, dst_id
-                );
-                return;
-            }
-        };
+        let next_id = self.network_service_mut().next_msg_id();
+        self.send_message_with_id(dst_id, message, next_id);
+    }
 
-        self.send_message_over_network(conn_info, message);
+    fn send_message_with_id(&mut self, dst_id: &PublicId, message: Message, msg_id: u64) {
+        self.send_message_with_id_to_targets(&[dst_id.clone()], 1, message, msg_id);
     }
 
-    fn send_message_over_network(&mut self, conn_info: ConnectionInfo, message: Message) {
+    fn send_message_to_targets(
+        &mut self,
+        dst_targets: &[PublicId],
+        dg_size: usize,
+        message: Message,
+    ) {
+        let next_id = self.network_service_mut().next_msg_id();
+        self.send_message_with_id_to_targets(dst_targets, dg_size, message, next_id);
+    }
+
+    fn send_message_with_id_to_targets(
```

Comment:
Looking at all the functions, I this we can remove all the `_with_id` functions, and have this function generate the id.
i.e:
send_direct_message call send_message
send_message call send_message_to_targets
send_message_to_targets internally call `let next_id = self.network_service_mut().next_msg_id();`

I don't think we are using the `_with_id` function otherwise?

Reply:
I was thinking they might be needed for resends... but since resends use `send_to_next_target`, which only uses `QuicP2p::send` directly, you may be right.